bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/PullJob.java

```diff
@@ -102,22 +102,25 @@ public void run() {
         try {
             response = asyncListener.get(HTTP_TIMEOUT_SECS, TimeUnit.SECONDS);
         } catch (InterruptedException e1) {
-            logger.warn("Download of calendar was interrupted.");
-            logger.debug("InterruptedException message is: {}", e1.getMessage());
+            logger.warn("Download of calendar was interrupted: {}", e1.getMessage());
+            request.abort(e1.getCause());
```

Comment:
Looking at the API doc, the cause must not be null.
If I am not wrong, e1.getCause() can be null. Maybe you can simply pass e1 ?
Same at the other places.